flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCLookupFunction.java

```diff
@@ -139,6 +139,14 @@ public void eval(Object... keys) {
 
 		for (int retry = 1; retry <= maxRetryTimes; retry++) {
 			try {
+				try {
+					if (!dbConn.isValid(connectionCheckTimeout)) {
+						establishConnectionAndPreparedStatement();
```

Comment:
Maybe you should close the connection and statement fisrt, and then reestablish.